pkg/networkserver/mac/link_adr.go

```diff
@@ -321,7 +322,23 @@ func HandleLinkADRAns(
 	}
 
 	ev := EvtReceiveLinkADRAccept
-	if !pld.ChannelMaskAck || !pld.DataRateIndexAck || !pld.TxPowerIndexAck {
+	rejected := false
+
+	// LoRaWAN 1.0.4 spec L534-538:
+	// An end-device SHOULD accept the channel mask controls present in LinkADRReq, even
+	// when the ADR bit is not set. The end-device SHALL respond to all LinkADRReq commands
+	// with a LinkADRAns indicating which command elements were accepted and which were
+	// rejected. This behavior differs from when the uplink ADR bit is set, in which case the end-
+	// device accepts or rejects the entire command.
+	if macspec.LinkADRReqRejected(macState.LorawanVersion) {
+		rejected = !pld.ChannelMaskAck ||
```

Comment:
But where in the spec does it differentiate the Dr and TX power acks from the channel mask ack, i.e. if ADR is disabled and channel mask is not acknowledged, that is not a rejection, right?